docs/new/references/flashblocks.md

```diff
@@ -0,0 +1,148 @@
+
+Here's how it works:
+
+1. The "sequencer" emits a flashblock every 200ms (so a maximum of 10 per block height)
+2. The instrumented Base node reader sends the increasing versions of the same block to the Substreams engine and eventually, the full block.
+3. To keep up with the chain, it may skip a few emissions of partial blocks, but will never send the transactions out-of-order.
+4. The Substreams engine will remember what was processed for each active Substreams and only process the new transactions since the last execution.
+5. It sends the PartialData for each part of the full block as it gets it from the flash blocks
+6. If there is a reorg, new and undo signals are sent for the full blocks, but there is no consideration for the sent partial blocks. The user must always consider that the partial blocks data may become invalid.
+7. For this reason, there is no "cursor" sent with the partial blocks data.
+
+#### Changes to Protobuf models
+
+* Partial blocks are not sent as `BlockScopedData`, but as `PartialBlockData`, which is new possible type for `Response.message`
+
+```proto
+message PartialBlockData {
+  MapModuleOutput output = 1;
+  sf.substreams.v1.Clock clock = 2;
+  uint32 partial_index = 3;
+}
+```
+
+* The `sf.substreams.rpc.v2.Blocks/Request`  and `sf.substreams.rpc.v3.Blocks/Request` now contain these parameters:
+
+``` proto
+// If true, partial blocks are also sent on the stream
+bool include_partial_blocks = 15;
+
+// If true, only partial blocks are sent, no 'block-scoped-data' or cursor.
+// This value supersedes include_partial_blocks
+bool partial_blocks_only = 16;
+```
+
+#### Example
+
+For the hypothetical scenario where 
+* a block #123 is being emitted as partial blocks
+* each partial block contains exactly 10 new transactions (to simplify the example)
+* the Substreams engine receives only the blocks with index 2, 4, 7 (some may be skipped to keep up with the chain HEAD)
+* finally, it receives the full block #123
+
+The module will be executed 4 times with partial data:
+  1. with transactions 0-20
+  2. with transactions 20-40
+  3. with transactions 40-70
+  4. with transaction 70-100 (when it gets the full block)
+  
+Then, the module will be executed again with the full block data. This is the data that will be used to apply changes to the stores, to ensure consistency before we execute the next blocks.
+      
+The user will receive:
+
+1. The result of execution of trx 0-20, within PartialBlockData with Clock(num=123, ID=0xaaaaaaaaa) and PartialIndex=2
+2. The result of execution of trx 20-40, within PartialBlockData with Clock(num=123, ID=0xbbbbbbbbbb) and PartialIndex=4
+3. The result of execution of trx 40-70, within PartialBlockData with Clock(num=123, ID=0xcccccccccc) and PartialIndex=7
+4. The result of execution of trx 70-100, within PartialBlockData with Clock(num=123, ID=0xdddddddddd) and PartialIndex=10
```

Comment:
Use some backticks here and there to improve general rendering, probably on `PartialBlockData` `Clock(...)` and `PartialIndex`